python/lsst/ctrl/bps/htcondor/htcondor_service.py

```diff
@@ -1270,44 +1284,61 @@ def _create_detailed_report_from_jobs(wms_workflow_id, jobs):
-        run_summary=_get_run_summary(dag_job),
-        state=_htc_status_to_wms_state(dag_job),
+        wms_id=f"{dag_ad['ClusterId']}.{dag_ad['ProcId']}",
+        global_wms_id=dag_ad.get("GlobalJobId", "MISS"),
+        path=dag_ad["Iwd"],
+        label=dag_ad.get("bps_job_label", "MISS"),
+        run=dag_ad.get("bps_run", "MISS"),
+        project=dag_ad.get("bps_project", "MISS"),
+        campaign=dag_ad.get("bps_campaign", "MISS"),
+        payload=dag_ad.get("bps_payload", "MISS"),
+        operator=_get_owner(dag_ad),
+        run_summary=_get_run_summary(dag_ad),
+        state=_htc_status_to_wms_state(dag_ad),
         jobs=[],
-        total_number_jobs=dag_job.get("total_jobs", total_jobs),
-        job_state_counts=dag_job.get("state_counts", state_counts),
+        total_number_jobs=dag_ad.get("total_jobs", total_jobs),
+        job_state_counts=dag_ad.get("state_counts", state_counts),
         exit_code_summary=_get_exit_code_summary(jobs),
     )
 
-    for job_id, job_info in jobs.items():
-        try:
-            job_report = WmsJobReport(
-                wms_id=job_id,
-                name=job_info.get("DAGNodeName", job_id),
-                label=job_info.get("bps_job_label", pegasus_name_to_label(job_info["DAGNodeName"])),
-                state=_htc_status_to_wms_state(job_info),
-            )
-            if job_report.label == "init":
-                job_report.label = "pipetaskInit"
-            report.jobs.append(job_report)
-        except KeyError as ex:
-            _LOG.error("Job missing key '%s': %s", str(ex), job_info)
-            raise
+    for job_id, job_ad in jobs.items():
+        if not is_service_job(job_id):
+            try:
+                job_report = WmsJobReport(
+                    wms_id=job_id,
+                    name=job_ad.get("DAGNodeName", job_id),
+                    label=job_ad.get("bps_job_label", pegasus_name_to_label(job_ad["DAGNodeName"])),
+                    state=_htc_status_to_wms_state(job_ad),
+                )
+                if job_report.label == "init":
+                    job_report.label = "pipetaskInit"
+                report.jobs.append(job_report)
+            except KeyError as ex:
+                _LOG.error("Job missing key '%s': %s", str(ex), job_ad)
+                raise
+        else:
+            job_label = job_ad.get("bps_job_label")
+            if job_label is None:
+                _LOG.warning("Service job with id '%s': missing label, no action taken", job_id)
+            elif job_label == dag_ad.get("bps_provisioning_job", "MISS"):
+                if report.state == WmsStates.RUNNING:
+                    report.specific_info = WmsSpecificInfo()
+                    job_state = _htc_status_to_wms_state(job_ad).name
+                    report.specific_info.add_message(
+                        template="Provisioning job status: {status}",
+                        context={"status": job_state},
+                    )
```

Comment:
Yes, that's correct. DAGMan deletes the service job when it exits and as a result this job's status is DELETED even if the workflow completed successfully. I was afraid that it may startle the users so decided to display provisioning job status only when the workflow is running.

I'll make `bps report` always show the status provisioning job. I'll see if I can easily turn the DELETED status of the service job to something less perplexing when the service job is cancelled by DAGMan or as a minimum document this feature.